pkg/v1/cli/signals/signals_test.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package signals
+
+import (
+	"sync"
+	"syscall"
+	"testing"
+	"time"
+)
+
+// TestSetupSignalHandler ensures that when a signal is sent, context channel
+// done is sent right away.
+func TestSetupSignalHandler(t *testing.T) {
+	tests := []struct {
+		desc      string
+		tolerance time.Duration
+		signal    syscall.Signal
+	}{
+		{
+			"Signal interrupt gets caught and sends context done",
+			time.Millisecond * 100,
+			syscall.SIGINT,
+		},
+		{
+			"Signal terminate gets caught and sends context done",
+			time.Millisecond * 100,
+			syscall.SIGTERM,
+		},
+	}
+
+	for _, tc := range tests {
+		ctx := SetupSignalHandler()
+
+		startTime := time.Now()
+		var wg sync.WaitGroup
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for {
+				select {
+				case <-ctx.Done():
+					return
+				default:
+					if time.Since(startTime) > tc.tolerance {
+						t.Errorf("%s: want: immediate interrupt signal, got: no signal after %v", tc.desc, tc.tolerance)
```

Comment:
Just use "subtests" with `t.Run` and you don't need to prefix `tc.desc` everywhere. See https://go.dev/blog/subtests or [one of the examples](https://github.com/vmware-tanzu/tanzu-framework/blob/e3dbe729ace6d139ede1118916e3024d0d20747c/pkg/v1/sdk/features/client/client_test.go#L53).